scripts/lib/CIME/case/check_input_data.py

```diff
@@ -6,19 +6,84 @@
 from CIME.XML.inputdata import Inputdata
 import CIME.Servers
 
-import glob
+import glob, hashlib, time
 
 logger = logging.getLogger(__name__)
+# The inputdata_checksum.dat file will be read into this hash if it's available
+chksum_hash = dict()
+local_chksum_file = 'inputdata_checksum.dat'
 
-def _download_if_in_repo(server, input_data_root, rel_path, isdirectory=False):
+def _download_checksum_file(server, input_data_root, chksum_file, user):
+    """
+    Return True if successfully downloaded
+    """
+    success = False
+    rel_path = chksum_file
+    full_path = os.path.join(input_data_root, local_chksum_file)
+    lockfile = _check_permissions_and_lock_inputdata_file(input_data_root, user)
```

Comment:
@billsacks here is the lock on the checksum file.

Reply:
The checksum file on local disk may be a combination of checksums from several servers and 
is available to any user once a user with write access creates it.  The --chksum option may currently be used without any access to the inputdata server or servers, it seems useful to me that you would be able to use this option when servers are not available.

Reply:
Thanks for pointing that out. But I still stand by my feeling that this checksum file should be in a case-specific location, rather than having the same file shared between all cases and users on the system. The more I think about this, the more strongly I feel about it: I feel that having a single file that is potentially written by all cases, for all users, is asking for trouble.

That said, if you still disagree, can we get a 3rd opinion here - from @jgfouca or someone else?

Reply:
(My response was meant in response to your first comment here, not your second. I'll think about your second comment for a few minutes.)

Reply:
> The checksum file on local disk may be a combination of checksums from several servers and
> is available to any user once a user with write access creates it. The --chksum option may currently be used without any access to the inputdata server or servers, it seems useful to me that you would be able to use this option when servers are not available.

I see your point, but I feel that the value of being able to run this when servers are unavailable does not justify the race condition issues caused by sharing a single file amongst all users and cases.